github3/github.py

```diff
@@ -51,13 +52,14 @@ class GitHub(GitHubCore):
         g = GitHub(user, password)
         g = GitHub(token=token)
         g = GitHub(user, token=token)
+        g = GitHub(token=token, isapp=True)
 
     This is simple backward compatibility since originally there was no way to
     call the GitHub object with authentication parameters.
     """
 
-    def __init__(self, username='', password='', token=''):
-        super(GitHub, self).__init__({}, self.new_session())
+    def __init__(self, username='', password='', token='', isapp=False):
+        super(GitHub, self).__init__({}, self.new_session(isapp))
```

Comment:
So now we have `self.new_session(isapp)` but we set the headers ourselves below in `login`. We need to pick one place to do this and do that correctly.